### PR TITLE
normalisation and toDouble() and fromDouble()

### PR DESCRIPTION
**Pull request by merliecat**
This pull request includes both the normalisation of DecimalFloats and the convertions to and from doubles.

### CODE REVIEW

File: artio-codecs/src/main/java/uk/co/real_logic/artio/fields/DecimalFloat.java [hunk 1]
```diff
@@ -189,4 +230,151 @@ public int compareTo(final DecimalFloat other)
...
+    private void setAndNormalise(final long value, final int scale)
+    {
+        this.value = value;
+        this.scale = scale;
+        normalise();
+    }
+
+    private void normalise()
+    {
+        final long valueCopy = value;
```
Comment (RichardWarburton) [inline artio-codecs/src/main/java/uk/co/real_logic/artio/fields/DecimalFloat.java]: You only use those copies in the ArithmeticException whose method references them. Shouldn't the arithemtic exception throw the new values and you reset the fields to the copy that you started with?

Perhaps an easier way to write this code would be to perform all the operations on local variables and then only set them to the fields at the end if they are inside the limits.

**Comment by RichardWarburton** [comment]
One comment aside, this looks good. Thanks a lot.

File: artio-codecs/src/main/java/uk/co/real_logic/artio/fields/DecimalFloat.java (hunk 1 above)
Comment (merliecat) [reply artio-codecs/src/main/java/uk/co/real_logic/artio/fields/DecimalFloat.java]: That's a fair comment.  I'll change to using local variables.

**Comment by merliecat** [comment]
Sorry it took a few days to get back to this.
I've commit the change you requested to use local variables in the normalise() method.
Regards.

**Comment by RichardWarburton** [comment]
Cheers @merliecat

**Pull request merged**